db.php

```diff
@@ -969,7 +969,7 @@ public function query( $query ) {
 			$elapsed = $this->timer_stop();
 			++$this->num_queries;
 
-			if ( preg_match( '/^\s*SELECT\s+SQL_CALC_FOUND_ROWS\s/i', $query ) ) {
+			if ( preg_match( '/^\s*SELECT\s+(DISTINCT\s+)?SQL_CALC_FOUND_ROWS\s/i', $query ) ) {
```

Comment:
If we want to be more strictish and support other query modifiers, this would work:
```suggestion
			if ( preg_match( '/^\s*SELECT\s+([A-Z_]+\s+)*SQL_CALC_FOUND_ROWS\s/i', $query ) ) {
```

Eg; `SELECT SOME_MODE DISTINCT SQL_CALC_FOUND_ROWS * FROM...`